src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -1882,8 +1883,7 @@ private static void EnsureVMInitialized()
         [System.Security.SecuritySafeCritical]
         internal static void NotifyWorkItemProgress()
         {
-            if (!ThreadPoolGlobals.vmTpInitialized)
-                ThreadPool.InitializeVMTp(ref ThreadPoolGlobals.enableWorkerTracking);
+            EnsureVMInitialized();
             NotifyWorkItemProgressNative();
         }
```

Comment:
Was the behaviour correct to not set `ThreadPoolGlobals.vmTpInitialized = true;` at this point previously?